Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp

```diff
@@ -361,12 +361,14 @@ void GStreamerMediaEndpoint::disposeElementChain(GstElement* element)
     auto pad = adoptGRef(gst_element_get_static_pad(element, "sink"));
     auto peer = adoptGRef(gst_pad_get_peer(pad.get()));
 
-    gst_element_set_locked_state(m_pipeline.get(), TRUE);
+    {
+        auto stateLocker = GstElementStateLocker(element);
+        gst_element_set_state(element, GST_STATE_NULL);
+    }
```

Comment:
This is totally ok, but I wonder, with this pattern, if we should create a function to just set the state while locked. Something like `gstElementLockAndSetState`.

Reply:
Yes, might be better indeed, i can rework this.